tests/test_diffeq/test_odefiltsmooth/test_initialize.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+import pytest
+
+import probnum.diffeq as pnde
+import probnum.filtsmooth.statespace as pnss
+import probnum.problems.zoo.diffeq as diffeq_zoo
+import probnum.random_variables as pnrv
+
+from ._known_initial_derivatives import LV_INITS, THREEBODY_INITS
+
+# Jax dependency handling
+# pylint: disable=unused-import
+try:
+    import jax
+    import jax.numpy as jnp
+    from jax.config import config
+
+    config.update("jax_enable_x64", True)
+
+    JAX_AVAILABLE = True
+
+except ImportError:
+    JAX_AVAILABLE = False
+
+
+# Pytest decorators to select tests for each case
+only_if_jax_available = pytest.mark.skipif(not JAX_AVAILABLE, reason="requires jax")
+
+
+@pytest.mark.parametrize("any_order", [0, 1, 2])
+@only_if_jax_available
+def test_compute_all_derivatives_via_taylormode(any_order):
+    """Test asserts that the examples in diffeq-zoo are compatible with
+    `compute_all_derivatives`, which happens if they are implemented in jax, and jax is
+    available in the current environment."""
+    ivp = diffeq_zoo.threebody_jax()
+    y0_all, errors = pnde.compute_all_derivatives_via_taylormode(
+        ivp.f, ivp.y0, ivp.t0, order=any_order
+    )
+    np.testing.assert_allclose(errors, 0.0)
+
+
+@pytest.fixture
+def order():
+    return 5
+
+
+@pytest.fixture
+def lv():
+    y0 = pnrv.Constant(np.array([20.0, 20.0]))
+
+    # tmax is ignored anyway
+    return pnde.lotkavolterra([0.0, None], y0)
+
+
+@pytest.fixture
+def lv_inits(order):
+    lv_dim = 2
+    vals = LV_INITS[: lv_dim * (order + 1)]
+    return pnss.convert_derivwise_to_coordwise(vals, ordint=order, spatialdim=lv_dim)
+
+
+def test_initialize_with_rk(lv, lv_inits, order):
```

Comment:
even though these kinds of tests are a bit messy currently, they provide the template for any sorts of tests for these implementations. If for instance we do some IEKS initialisation at some point, it can be tested just like this here.